docs/pallas/gpu/reference.md

```diff
@@ -162,6 +307,18 @@ There are three operations that can complete a barrier:
   Failing to satisfy this will corrupt the data structure and can cause surprising failures
   (including CUDA runtime errors). See below for an example of a valid program with two threads.
 
+> Another critical restriction is that the number of barrier completions must equal the
+  number of barrier waits throughout its lifetime. It is not allowed to end a scoped
```

Comment:
nit: its -> [the barrier's]? (it's not been mentioned as a "proper" subject yet :))